graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphQLService.java

```diff
@@ -0,0 +1,226 @@
+                                                 @Nullable Map<String, Object> variables,
+                                                 @Nullable String operationName) {
+        final ExecutionInput.Builder builder = ExecutionInput.newExecutionInput(query);
+        if (variables != null && !variables.isEmpty()) {
+            builder.variables(variables);
+        }
+        if (operationName != null) {
+            builder.operationName(operationName);
+        }
+        return builder.context(ctx)
+                      .dataLoaderRegistry(dataLoaderRegistry)
+                      .build();
+    }
+
+    private static Map<String, Object> toVariableMap(@Nullable Object variables) {
+        if (variables == null) {
+            return ImmutableMap.of();
+        }
+
+        if (variables instanceof Map) {
+            final Map<?, ?> variablesMap = (Map<?, ?>) variables;
+            final ImmutableMap.Builder<String, Object> builder =
+                    ImmutableMap.builderWithExpectedSize(variablesMap.size());
+            variablesMap.forEach((k, v) -> builder.put(String.valueOf(k), v));
+            return builder.build();
+        } else {
+            throw new IllegalArgumentException("Unknown parameter type variables");
+        }
+    }
+
+    private static Map<String, Object> toVariableMap(@Nullable String value) {
+        if (Strings.isNullOrEmpty(value)) {
+            return ImmutableMap.of();
+        }
+        return parseJsonString(value);
+    }
+
+    private HttpResponse execute(ServiceRequestContext ctx, ExecutionInput input) {
+        if (useBlockingTaskExecutor) {
+            final CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+            ctx.blockingTaskExecutor().execute(() -> {
+                try {
+                    final ExecutionResult executionResult = graphQL.execute(input);
+                    future.complete(toHttpResponse(executionResult));
+                } catch (Throwable e) {
+                    final ExecutionResult error = newExecutionResult(e);
+                    future.complete(HttpResponse.of(MediaType.JSON_UTF_8,
+                                                    toJsonString(error.toSpecification())));
+                }
+            });
+            return HttpResponse.from(future);
+        }
+
+        return HttpResponse.from(graphQL.executeAsync(input).handle((executionResult, cause) -> {
+            if (cause != null) {
+                final ExecutionResult error = newExecutionResult(cause);
+                return HttpResponse.of(MediaType.JSON_UTF_8, toJsonString(error.toSpecification()));
+            }
+            return toHttpResponse(executionResult);
+        }));
+    }
+
+    private static HttpResponse toHttpResponse(ExecutionResult executionResult) {
+        // TODO: When WebSocket is implemented, it should be removed.
+        if (executionResult.getData() instanceof Publisher) {
```

Comment:
Let's leave a log here.
`logger.warn("executionResult.getData() returns a {} that is not supported yet.", ...)`